src/activities/content/ContentEntity.js

```diff
@@ -36,4 +39,13 @@ export class ContentEntity extends Entity {
 			&& this._entity.hasLinkByRel(Rels.Content.moduleEntity)
 			&& this._entity.getLinkByRel(Rels.Content.moduleEntity).href;
 	}
+
+	/**
+	 * @returns {string} content-weblink link
+	 */
+	getWebLinkHref() {
+		return this._entity
+			&& this._entity.hasLinkByRel(Rels.Content.weblinkEntity)
+			&& this._entity.getLinkByRel(Rels.Content.weblinkEntity).href;
+	}
```

Comment:
So we get a `ContentEntity` when we get the activity usage and have to dig a layer deeper to get the actual information for either modules or weblinks by fetching the appropriate link. All the currently defined entities can be seen used in the `content.js` state file in activities [here](https://github.com/BrightspaceHypermediaComponents/activities/blob/master/components/d2l-activity-editor/d2l-activity-content-editor/state/content.js). This state file gets the `ContentEntity` [here](https://github.com/BrightspaceHypermediaComponents/activities/blob/master/components/d2l-activity-editor/d2l-activity-content-editor/state/content.js#L35) and then loads the relevant information depending on the type of content. If it is a module, it proceeds to load the `ContentModuleEntity` [here](https://github.com/BrightspaceHypermediaComponents/activities/blob/master/components/d2l-activity-editor/d2l-activity-content-editor/state/content.js#L44).

It is known that this state file will need to be further separated so that it doesn't get too complex with the addition of more entity types but that is the basic flow that I have currently setup. It didn't make sense to me to have `ContentEntity` as a superclass because all of the defined functions in `ContentEntity` would not work on `ContentModuleEntity` or `ContentWebLinkEntity`.